### PR TITLE
add next steps to service entity

### PR DESCRIPTION
**Pull request by cristinabuenahora**
## Change description

Shows next steps 
Screen Shot 2022-02-23 at 9 19 07 PM

If there's no ladder, it remains the same and shows rank
Screen Shot 2022-02-23 at 9 19 14 PM

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/components/EntityPage/EntityScorecardDetails/EntityScorecardDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Cortex Applications, Inc.
```
Comment (nikhilunni) [inline src/components/EntityPage/EntityScorecardDetails/EntityScorecardDetails.tsx:2]: super nit: idk why the eslint-config didn't update to 2022, could we update these?

File: src/components/Scorecards/ScorecardsServicePage/ScorecardsServiceNextRules.tsx [hunk 2]
```diff
@@ -77,22 +84,20 @@ export const ScorecardsServiceNextRules = ({
...
+          {nextSteps.nextLevel !== undefined ? (
+            <Typography variant="subtitle1" className={classes.level}>
+              <b>Next Level: </b> {nextSteps.nextLevel.name}
+              <LoyaltyIcon style={{ color: `${nextSteps.nextLevel.color}` }} />
+              <br />
+              Complete these rules to get to the next level:
+              <br />
+              {nextSteps.rulesToComplete.map(rule => (
+                <i key={`NextRule-${rule.id}`}>
+                  &#8226; {rule.title ?? rule.expression}
```
Comment (nikhilunni) [inline src/components/Scorecards/ScorecardsServicePage/ScorecardsServiceNextRules.tsx:96]: there's a utility for this 😄 `ruleName` just to make sure we don't miss it anywhere

**Review by nikhilunni — APPROVED** [review]
lgtm!

**Comment by gsdatta** [comment]
minor nit, can we add `hr` or some kind of divider between current level, next level, and what's next to create a similar visual separation like in the Cortex ui

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardsTableCard/RuleResultDetails.tsx [hunk 3]
```diff
@@ -44,10 +48,19 @@ export const RuleResultDetails = ({
...
+
+  const [open, setOpen] = useState(false);
+  const showExpandButton = rule.rule.description || rule.rule.title;
+  const hasTitle = !!rule.rule.title;
   const isFailing = rule.score === 0;
-  
+
   return (
     <ListItem alignItems="flex-start">
+      {showExpandButton && (
```
Comment (nikhilunni) [inline src/components/Scorecards/ScorecardDetailsPage/ScorecardsTableCard/RuleResultDetails.tsx:59]: played around with this for a bit but any way we can get these to be centered?
image
maybe even abandoning the `List` component altogether and just using the raw `Grid` (even outside in `ScorecardResultDetails`? if it's taking too long no worries, just a small nitpick

**Review by nikhilunni — APPROVED** [review]
lgtm otherwise! all the entityrefs are working as expected

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardsTableCard/ScorecardsTableRow.tsx [hunk 4]
```diff
@@ -59,35 +50,28 @@ export const ScorecardsTableRow = ({
...
-                componentRef={score.componentRef}
-              >
+            <Box
+              flexDirection="row"
+              display="flex"
+              justifyContent="flex-start"
+              alignItems="center"
+            >
+              <Box alignSelf="center" width={1 / 10}>
+                <Gauge
```
[the remark this answers is not in the document]
Comment (cristinabuenahora) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardsTableCard/ScorecardsTableRow.tsx:68]: fixed.

**Review by nikhilunni — APPROVED** [review]

**Pull request merged**